From cd44ee903a04e05c07cdc4badd33f0e719ff5151 Mon Sep 17 00:00:00 2001
From: Fam Zheng <famz@redhat.com>
Date: Mon, 4 Aug 2014 09:48:44 -0400
Subject: [virt] hyperv/storvsc: Set cmd_per_lun to reflect value supported by the Host

Message-id: <1407145729-25017-3-git-send-email-famz@redhat.com>
Patchwork-id: 86953
O-Subject: [RHEL-7 PATCH 2/7] Drivers: scsi: storvsc: Set cmd_per_lun to reflect value supported by the Host
Bugzilla: 1122317
RH-Acked-by: Jason Wang <jasowang@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

From: K. Y. Srinivasan <kys@microsoft.com>

Set cmd_per_lun to reflect value supported by the Host.

Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Reviewed-by: Hannes Reinecke <hare@suse.de>
Cc: <stable@vger.kernel.org>
Signed-off-by: Christoph Hellwig <hch@lst.de>
(cherry picked from commit 52f9614dd8294e95d2c0929c2d4f64b077ae486f)
Signed-off-by: Fam Zheng <famz@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/storvsc_drv.c b/drivers/scsi/storvsc_drv.c
index 64b3bac..1fdb47f 100644
--- a/drivers/scsi/storvsc_drv.c
+++ b/drivers/scsi/storvsc_drv.c
@@ -1687,7 +1687,7 @@ static struct scsi_host_template scsi_driver = {
  .slave_alloc =  storvsc_device_alloc,
  .slave_destroy = storvsc_device_destroy,
  .slave_configure = storvsc_device_configure,
- .cmd_per_lun =  1,
+ .cmd_per_lun =  255,
  .can_queue =  STORVSC_MAX_IO_REQUESTS*STORVSC_MAX_TARGETS,
  .this_id =  -1,
  /* no use setting to 0 since ll_blk_rw reset it to 1 */
-- 
1.7.1